cpu/sam0_common/sam0_eth/eth-netdev.c

```diff
@@ -140,24 +141,24 @@ void sam0_eth_setup(netdev_t* netdev)
     netdev_register(netdev, NETDEV_SAM0_ETH, 0);
 }
 
-/* TODO: rework the whole isr management... */
 void isr_gmac(void)
 {
     uint32_t isr;
-    uint32_t tsr;
     uint32_t rsr;
 
     isr = GMAC->ISR.reg;
-    tsr = GMAC->TSR.reg;
     rsr = GMAC->RSR.reg;
+    (void)isr;
 
     if (rsr & GMAC_RSR_REC) {
         netdev_trigger_event_isr(_sam0_eth_dev.netdev);
     }
-
-    GMAC->TSR.reg = tsr;
+    if (rsr & GMAC_RSR_BNA) {
```

Comment:
Better add a comment as to what that flag means and when it occurs

Reply:
Done.